tests/integration_tests/data_steward/cdr_cleaner/cleaning_rules/generalize_state_by_population_test.py

```diff
@@ -0,0 +1,157 @@
+        Add data to the tables for the rule to run on.
+
+        :param sql_statements: a list of sql statements to load for testing.
+        """
+        pandas_gbq.to_gbq(df,
+                          f'{dataset_id}.{table}',
+                          project_id=project_id,
+                          if_exists='replace')
+
+    def test_generalize_state_by_population_cleaning(self):
+        """
+        Tests that the specifications for STATE_GENERALIZATION_QUERY perform as designed.
+
+        Validates pre conditions, tests execution, and post conditions based on the load
+        statements and the tables_and_counts variable.        
+        """
+
+        #Rows to insert
+        inserted_rows = []
+        test_obs_id = 1
+        for test_state in TEST_STATES.values():
+            name = test_state['name']
+            concept_id = test_state['concept_id']
+            participant_count = test_state['participant_count']
+
+            for i in range(participant_count):
+                row = {
+                    'observation_id': test_obs_id,
+                    'person_id': test_obs_id,
+                    'observation_concept_id': 40766229,
+                    'observation_date': self.date,
+                    'observation_type_concept_id': 45905771,
+                    'value_as_concept_id': 1,
+                    'observation_source_concept_id': 1585249,
+                    'value_source_value': name,
+                    'value_source_concept_id': concept_id
+                }
+
+                inserted_rows.append(row)
+                test_obs_id += 1
+        inserted_rows_df = pd.DataFrame(inserted_rows,
+                                        columns=list(inserted_rows[0].keys()))
+
+        #Rows to expect
+        expected_rows = inserted_rows
+        for i, expected_row in enumerate(expected_rows):
+            if TEST_STATES[expected_row['value_source_value']][
+                    'participant_count'] < PARTICIPANT_THRESH:
+                expected_rows[i].update({
+                    'value_source_concept_id': 2000000011,
+                    'value_as_concept_id': 2000000011
+                })
+        expected_rows_df = pd.DataFrame(expected_rows,
+                                        columns=list(expected_rows[0].keys()))
+        expected_rows_df['observation_date'] = '2020-05-05'
+
+        #Insert rows
+        self.load_test_data(inserted_rows_df, self.project_id, self.dataset_id,
+                            'observation')
+
+        #Check match
+        tables_and_counts = [{
+            'fq_table_name': '.'.join([self.fq_dataset_name, 'observation']),
+            'fq_sandbox_table_name': '',
+            'loaded_ids': list(expected_rows_df['observation_id']),
+            'sandboxed_ids': [],
+            'fields': list(expected_rows_df.columns),
+            'cleaned_values': [tuple(row) for row in expected_rows_df.values]
+        }]
+
+        self.default_test(tables_and_counts)
```

Comment:
nit: missing newline at end of file. Usually good style to add it for all files.

Reply:
Sure, I'll handle this and the other suggestions.